lib/block-supports/colors.php

```diff
@@ -142,6 +141,31 @@ function gutenberg_apply_colors_support( $block_type, $block_attributes ) {
 	return $attributes;
 }
 
+/**
+ * Checks whether serialization of the current block's color properties
+ * should occur.
+ *
+ * @param WP_Block_type $block_type Block type.
+ * @param string        $feature    Optional name of individual feature to check.
+ *
+ * @return boolean Whether to serialize color support styles & classes.
+ */
+function gutenberg_skip_color_serialization( $block_type, $feature = null ) {
+	if ( ! is_object( $block_type ) ) {
```

Comment:
The other skip serialization methods don't have this guard? Rather than add it to all of them is it possible to make this a shared method and pass in `$path` as that seems to be the only difference between them?

Reply:
Good point @glendaviesnz 

We're doing it in the JS: https://github.com/WordPress/gutenberg/pull/36293/files#diff-cd3303ad473e0bbfc9d92758c62c8f4295b31a0bb8283a3e4445c38fb73c0f4dR106

I probably should have picked up on that. 

So maybe we could have a `block-supports/utils.php` file or something with the following:

```php
/**
 * Checks whether serialization of the current block's typography properties
 * should occur.
 *
 * @param WP_Block_type $block_type  Block type.
 * @param string        $feature_set Name of block support feature set..
 * @param string        $feature     Optional name of individual feature to check.
 *
 * @return boolean Whether to serialize typography support styles & classes.
 */
function gutenberg_should_skip_block_supports_serialization( $block_type, $feature_set, $feature = null ) {
	if ( ! $block_type || ! $feature_set ) {
		return false;
	}

	$path               = array( $feature_set, '__experimentalSkipSerialization' );
	$skip_serialization = _wp_array_get( $block_type->supports, $path, false );

	if ( is_array( $skip_serialization ) ) {
		return in_array( $feature, $skip_serialization, true );
	}

	return $skip_serialization;
}
```

Reply:
> The other skip serialization methods don't have this guard? Rather than add it to all of them is it possible to make this a shared method and pass in $path as that seems to be the only difference between them?

I've done this in 9e380f0652d60833e4d8ec80fa493ad75256c76d

Thanks for the suggestion @glendaviesnz 

I smoke tested and reran all unit tests... things seem to work well, but ran out of time to retest the individual supports manually.

Reply:
This change looks good to me, and when tested against the group block none of the color, spacing, border, typography attributes were serialised onto the block wrapper.